hibernate-types-52/src/main/java/com/vladmihalcea/hibernate/type/money/MonetaryAmountType.java

```diff
@@ -75,14 +75,14 @@ public int hashCode(Object x) throws HibernateException {
 
     @Override
     public Object nullSafeGet(ResultSet rs, String[] names, SharedSessionContractImplementor session, Object owner) throws HibernateException, SQLException {
-        if (rs.wasNull()) {
-            return null;
-        }
-
         String amountColumnName = names[0];
         String currencyColumnName = names[1];
 
         BigDecimal amount = rs.getBigDecimal(amountColumnName);
+        if(amount == null) {
+            return null;
+        }
+
         String currency = rs.getString(currencyColumnName);
 
```

Comment:
Null-check should be here. Because creating monetary type without currency make no sense and could rise an exception `Currency Code may not be null`.

I propose to use here:
```
if (rs.wasNull()) {
    return null;
}
```

Reply:
Question is should the system thow an exception if only one of them is null or still ignored it an return also null in that case. I`m was not sure about this case and decide to let the exception be thrown.

Is there a diffferend if i use `rs.wasNull()` or check the value of the last read column?
To get the correct result of `rs.wasNull()` i need to read the column before. It also only checks if the last read column is null.

Reply:
Because the Java doc of the `rs.getBigDicimal(x)` says: `if the value is SQL NULL, the value returned is null`

Reply:
`null` seems to be enough.

Reply:
But we should reconsider strategy. If any of a column has `null` - that should be `null` or exception? Maybe if any value is `null` - `null` should be returned.